### PR TITLE
[Bugfix][MLA] Do not use Dense MHA for GLM-5.2

### PR DESCRIPTION
**Pull request by WoosukKwon**
## Purpose

Fix incorrect short-prefill dispatch in the NVIDIA DeepSeek-V3.2 attention
wrapper.

The generic sparse-MLA metadata path may select dense MHA when a prefill's
sequence length is at most `index_topk`. It then allows the sparse indexer to
skip top-k scoring. `DeepseekV32Attention`, however, only invokes
`forward_mqa`; it does not execute the dense-MHA backend. For a pure prefill
above the backend's decode-like threshold, MQA therefore consumed the cleared
all-`-1` top-k buffer and GLM-5.2 output degenerated into repeated tokens.

This change adds a model-layer dense-prefill capability in addition to the
backend capability. The NVIDIA DeepSeek-V3.2 wrapper declares that it is
MQA-only, so sparse MLA keeps its prefill on top-k MQA and the indexer does not
advertise the dense-MHA scoring skip.

This does not duplicate the open related work:

- #51395 disables dense prefill only for the SM120 sparse-MLA backend.
- #52046 adds a dense branch for the PCP path; the reproduced failure is the
  non-PCP SM100 path.
- #49790 routes DSA architectures to the NVIDIA implementation but does not
  resolve this dispatch mismatch.

Issue #46715 reports a different GLM-5.1 failure at a 65,536-token scheduler
budget and multi-thousand-token prompts on the registered model path.

## Test Plan

```text
.venv/bin/python -m pytest \
  tests/model_executor/layers/test_mla_short_prefill_indexer.py -q

pre-commit run --files \
  vllm/model_executor/layers/attention/mla_attention.py \
  vllm/models/deepseek_v32/attention.py \
  tests/model_executor/layers/test_mla_short_prefill_indexer.py
```

Real-model validation uses `nvidia/GLM-5.2-NVFP4`, TP=4, the explicit NVIDIA
DeepSeek-V3.2 class override, FP8 E4M3 KV cache, eager execution, FP32 indexer
logits, GVR disabled, and FlashInfer autotuning disabled. It serially probes
the 128/129-token routing boundary without setting
`sparse_mla_force_mqa=true`.

## Test Result

- Focused pytest: `7 passed`.
- Pre-commit on all three changed files: all hooks passed.
- `compileall` and `git diff --check`: passed.
- Before this change, the real model passed through 127 actual prompt tokens
  and failed at 129 and every tested larger prompt through 1,001 tokens. Adding
  only `sparse_mla_force_mqa=true` made all 19 boundary cases pass, causally
  confirming the routing mismatch.
- Patched real-model result: all 19/19 cases passed exact retrieval without the
  force-MQA workaround, spanning 97, 101, 122, 127, 129, 130, 131, 132, 162,
  191, 192, 193, 194, 196, 202, 257, 301, 502, and 1,001 actual prompt tokens.

AI assistance was used to diagnose, implement, test, and draft this change.
The human submitter must review every changed line and understand and defend
the change end-to-end before marking the PR ready for review.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR is described.
- [x] The test plan includes exact commands.
- [x] Unit and real-model results are reported.
- [x] Related open work was checked for duplication.
</details>

### CODE REVIEW

**Comment by WoosukKwon** [comment]
This PR essentially reverts #51298. I chose to just disable dense MHA instead of adding support for it, since the complexity outweighs the performance gain.
